plugins/moveToArchive.koplugin/main.lua

```diff
@@ -0,0 +1,139 @@
+            {
+                text = _("Go to last copied/moved from folder"),
+                callback = function()
+                    if not last_copied_from_dir then
+                        UIManager:show(InfoMessage:new{
+                            text = _("No previous folder found.")
+                         })
+                        return
+                    end
+                    if FileManager.instance then
+                        FileManager.instance:reinit(last_copied_from_dir)
+                    else
+                        FileManager:showFiles(last_copied_from_dir)
+                    end
+                end,
+            },
+            {
+                text = _("Set archive directory"),
+                keep_menu_open = true,
+                callback =  self.setArchiveDirectory,
+            }
+        },
+    }
+end
+
+function MoveToArchive:moveToArchive()
+    local move_done_text = _("Book moved. \nDo you want to open it from archive folder?")
+    local doc_old_path = MoveToArchive:commonProcess(true, move_done_text)
+    ReadHistory:removeItemByPath(doc_old_path)
+end
+
+function MoveToArchive:copyToArchive()
+    local copy_done_text =_("Book copied. \nDo you want to open it from archive folder?")
+    MoveToArchive:commonProcess(false, copy_done_text)
+end
+
+function MoveToArchive:commonProcess(is_move_process, moved_done_text)
+    if not archive_dir_path then
+        MoveToArchive:showNoArchiveConfirmBox()
+        return
+    end
+    local document_full_path = G_reader_settings:readSetting("lastfile")
+    local filename
+    last_copied_from_dir, filename = util.splitFilePathName(document_full_path)
+
+    logger.dbg("MoveToArchive: last_moved/copied_from_dir :", last_copied_from_dir)
+    move_to_archive_settings:saveSetting(last_copied_from_config_key, ("%s/"):format(last_copied_from_dir))
+
+    if is_move_process then
+        FileManager:moveFile(document_full_path, archive_dir_path)
+    else
+        FileManager:copyFileFromTo(document_full_path, archive_dir_path)
+    end
+
+    MoveToArchive:showConfirmBox(moved_done_text, function () ReaderUI:showReader(archive_dir_path .. filename) end)
+    return document_full_path
+end
+
+
+function MoveToArchive:setArchiveDirectory()
+    require("ui/downloadmgr"):new{
+        onConfirm = function(path)
+            logger.dbg("MoveToArchive: set archive directory to: ", path)
+            move_to_archive_settings:saveSetting(archive_dir_config_key, ("%s/"):format(path))
+            move_to_archive_settings:flush()
+
+            archive_dir_path = path
+        end,
+    }:chooseDir()
+end
+
+function MoveToArchive:showNoArchiveConfirmBox()
+    MoveToArchive:showConfirmBox(_("No archive directory. \nDo you want to set it now?"), self.setArchiveDirectory)
+end
+
+function MoveToArchive:showConfirmBox(text, ok_callback)
+    UIManager:show(ConfirmBox:new{
+        text = text,
+        cancel_text = _("Cancel"),
+        ok_text = _("Ok"),
```

Comment:
That's the default, but it should describe the action to be confirmed, see http://koreader.rocks/doc/modules/ui.widget.confirmbox.html
```suggestion
        ok_text = _("Set archive folder"),
```